src/prommis/nanofiltration/membrane_cascade_flowsheet/solute_property.py

```diff
@@ -0,0 +1,129 @@
+"""Generic multi-solute property package for membrane filtration flowsheet."""
+
+import pyomo.environ as pyo
+from idaes.core import (
+    Component,
+    declare_process_block_class,
+    MaterialFlowBasis,
+    Phase,
+    PhysicalParameterBlock,
+    StateBlock,
+    StateBlockData,
+)
+from idaes.core.util.initialization import fix_state_vars
+from pyomo.common.config import ConfigValue
+from idaes.core.util.exceptions import ConfigurationError
+
+
+class _StateBlock(StateBlock):
+    def fix_initialization_states(self):
+        """
+        Fix state variables for state blocks.
+
+        Returns:
+            None
+        """
+        fix_state_vars(self)
+
+    def initialization_routine(self):
+        pass
+
+
+@declare_process_block_class("SoluteStateBlock", block_class=_StateBlock)
+class SoluteStateBlock1Data(StateBlockData):
+    """Defines thermophysical property variables and constraints."""
+
+    def build(self):
+        """Construct model variables/expressions/constraints."""
+        super().build()
+
+        self.flow_vol = pyo.Var(
+            units=pyo.units.m**3 / pyo.units.hour,
+            bounds=(1e-8, None),
+        )
+
+        # remove solvent from component list
+        solutes = [i for i in self.component_list if i != 'solvent']
+
+        self.mass_solute = pyo.Var(
+            solutes,
+            units=pyo.units.kg / pyo.units.hour,
+            bounds=(1e-8, None),
+        )
+
+    def get_material_flow_terms(self, p, j):
+        """Get mass flow rate."""
+        if j == "solvent":
+            # Assume constant density of pure water
+            # return self.flow_vol * self.params.dens_H2O
```

Comment:
The density term should be cancelled out in all water relevant mass balances and membrane constraints. This was removed way to help with scaling (all water mass flows were much higher than solute mass flows).

Would it be preferred to keep the density term for correctness and manually scale the water related constraints?

Reply:
Yes, I think it should include the density term for completeness and then add a scaling factor for `self.flow_vol`.

Reply:
I have opted to scale all constraints that include the density term rather than directly scaling the variables to avoid dealing with interpretability issues (i.e. having very small `flow_vol` terms that need to be scaled back after solving).

The relevant method has been added to the `DIafiltrationModel` class initialization routine as a `model_scaling` function in [this commit](https://github.com/prommis/prommis/pull/104/commits/51b7824f7a2023ab4d28a6dd31a8de50ae0413ad).

Reply:
This seems like a bad idea to me. Firstly, this looks like it would cause unit consistency issues with the other terms in the material balances as they will expect the units defined in the metadata. As far as scaling goes, that is what the scaling tools are for, and with the new Pyomo solver interfaces all the scaling and unscaling is handled automatically.

Even if the unit consistency is not an issue, the fact that this is doing something non-standard will likely cause someone confusion later on.,

Reply:
Hi Andrew, thanks for the feedback on this. Are you referring to the removal of the density term from the property package? This has been addressed, and I added back the density term. 

I am now using scaling tools on relevant constraints containing the density term in [this commit](https://github.com/prommis/prommis/pull/104/commits/51b7824f7a2023ab4d28a6dd31a8de50ae0413ad) and wanted to include a comment on it.

Reply:
I was primarily referring to the density term - it was not clear to me from the comment that you had put it back in.

For the scaling part, you are using the old scaling tools which are in the process of being deprecated. You should switch to the new scaling tools in `idaes.core.scaling` (instead of `idaes.core.util.scaling`) and look at making use of the standard workflow (using `Scaler` classes). This might be something for another PR however.

Reply:
Thanks! I'll look at updating to the new scaling tools.
Apologies for the confusion. I'll resolve this conversation since the primary issue has been addressed.